tests/edgeless.spec.ts

```diff
@@ -282,9 +283,54 @@ test('selection box of shape element sync on fast dragging', async ({
   const box = await page
     .locator('.affine-edgeless-selected-rect')
     .boundingBox();
-  if (!box) {
-    throw new Error('box is null');
-  }
-  const { x, y, width, height } = box;
-  expect([x, y, width, height]).toStrictEqual([650, 450, 100, 100]);
+  assertHoverRect(box, ['x', 'y', 'width', 'height'], [650, 450, 100, 100]);
+});
+
+test('hover state for shape element', async ({ page }) => {
+  await enterPlaygroundRoom(page);
+  await initEmptyEdgelessState(page);
+  await switchEditorMode(page);
+
+  await switchMouseMode(page);
+  await dragBetweenCoords(page, { x: 100, y: 100 }, { x: 200, y: 200 });
+  await switchMouseMode(page);
+
+  await page.mouse.move(150, 150);
+
+  const hoverRect = page.locator('.affine-edgeless-hover-rect');
+  const box = await hoverRect.boundingBox();
+
+  assertHoverRect(box, ['x', 'y', 'width', 'height'], [100, 100, 100, 100]);
```

Comment:
Is this okay?

Reply:
I think `assertHoverRect(page, box, 100, 100, 100, 100)` should make more sense? And this util should also handle the `page.locator('.affine-edgeless-hover-rect')` setup.

Reply:
In this mode `assertHoverRect(page, box, 100, 100, 100, 100)`, `box` is a selector for `page.locator(box)`?

Reply:
if this function just for 'hover rect', we can use this `assertHoverRect(page, 100, 100, 100, 100)`

Reply:
done :)